components/submissions/StatusBadge.tsx

```diff
@@ -0,0 +1,20 @@
+import { FC } from 'react';
+import { tw } from '../../utils';
+
+interface Props {
+    status: string
+}
+
+const possibleStatus: {[key: string]: {color: string, text: string}} = {
```

Comment:
when defining objects you can let typescript know that it will not have any changes done to it. To do that you can do:

```
const possibleStatus = {
  'pending': {color: 'bg-blue-100 text-blue-700', text: 'Pending'},
  'in_progress': {color: 'bg-green-100 text-green-700', text: 'In Progress'},
  'done': {color: 'bg-gray-100 text-gray-700', text: 'Done'},
} as const;
```

That way not only do you not need to specify the type of possibleStatus, you also get the benefit of having the exact keys of the object defined in the type